lib/rubocop/cop/rails/save_bang.rb

```diff
@@ -221,15 +221,21 @@ def check_used_in_conditional(node)
           true
         end
 
-        def conditional?(node) # rubocop:disable Metrics/CyclomaticComplexity
+        def in_condition_or_compound_boolean?(node)
           node = node.block_node || node
+          parent = node.parent
+          return false unless parent
 
-          condition = node.parent
-          return false unless condition
+          operator_or_single_negative?(parent) ||
+            (conditional?(parent) && node == parent.condition)
+        end
+
+        def operator_or_single_negative?(node)
+          node.or_type? || node.and_type? || single_negative?(node)
+        end
 
-          condition.if_type? || condition.case_type? ||
-            condition.or_type? || condition.and_type? ||
-            single_negative?(condition)
+        def conditional?(parent)
+          parent.if_type? || parent.case_type? || parent.while_type?
```

Comment:
Can you add a test case that requires `parent.while_type?`?
I think this change is a different reason for this PR, so it's easier to review it if you change it by opening another PR.

Reply:
I'll remove the `while_type` part